### PR TITLE
[sdk-js] Update to use the new search service #2900

### PR DESCRIPTION
**Pull request by jvega190**

### CODE REVIEW

File: packages/classes/src/config.ts [hunk 1]
```diff
@@ -31,7 +31,8 @@ const DEFAULTS: CrafterConfig = {
     GET_NAV_TREE: '/api/1/site/navigation/tree.json',
     GET_BREADCRUMB: '/api/1/site/navigation/breadcrumb.json',
     TRANSFORM_URL: '/api/1/site/url/transform.json',
-    SEARCH: 'crafter-search/api/2/search/search.json'
+    SEARCH: 'crafter-search/api/2/search/search.json',
+    ENGINE_SEARCH: 'api/1/site/elasticsearch/search'
```
Comment (sumerjabri) [inline packages/classes/src/config.ts]: Please rename to `Elasticsearch`. Make sure it's one word. Like this: `ELASTICSEARCH`.

Are you sure you have the right URL?

File: packages/search/README.md [hunk 2]
```diff
@@ -28,7 +28,35 @@ Map model
...
+
+  //First, set the Crafter configuration to _cache_ your config. 
+  //All subsequent calls to `getConfig` will use that configuration.
+  crafterConf.configure({
+    baseUrl: 'http://localhost:8080'
+    site: 'editorial'
+  });
+
+  //Create elastic query
+  const query = SearchService.createQuery('elastic');
```
Comment (sumerjabri) [inline packages/search/README.md]: `elasticsearch`

File: packages/search/README.md [hunk 3]
```diff
@@ -58,9 +87,12 @@ You may alternatively use a different config by supplying the config object at t
   import { SearchService } from '@craftercms/search';
 
   //Create query
-  const query = SearchService.createQuery();
-  query.query = "*:*";
-  query.filterQueries = ['content-type:"/component/video"'];
+  const query = SearchService.createQuery('elastic');
```
Comment (sumerjabri) [inline packages/search/README.md]: elasticsearch

Comment (jvega190) [reply packages/search/README.md]: done

File: packages/search/README.md (hunk 2 above)
Comment (jvega190) [reply packages/search/README.md]: done

File: packages/classes/src/config.ts (hunk 1 above)
Comment (jvega190) [reply packages/classes/src/config.ts]: I checked with Jose R. and it seems fine

File: packages/search/src/search-service.ts [hunk 4]
```diff
@@ -41,25 +42,38 @@ export class SearchService extends SDKService {
...
+      requestURL = composeUrl(config, crafterConf.getConfig().endpoints.ELASTICSEARCH) + '?crafterSite=' + config.site;
+
+      return SDKService.httpPost(requestURL, params)
+        .pipe(map((response: any) => {
+          return response.hits;
+        }));
+    } else {
+      requestURL = composeUrl(config, crafterConf.getConfig().endpoints.SEARCH);
+
+      for (let param in params) {
```
Comment (rart) [inline packages/search/src/search-service.ts]: We should do `params.hasOwnProperty` right inside the for body

File: packages/search/src/search-service.ts [hunk 5]
```diff
@@ -41,25 +42,38 @@ export class SearchService extends SDKService {
...
             ? queryOrParams.params
             : queryOrParams,
           searchParams = new URLSearchParams();
 
-    for (let param in params) {
-      if (Array.isArray(params[param])) {
-        for (let x = 0; x < params[param].length; x++) {
-          searchParams.append(param, params[param][x]);
+    if (queryOrParams instanceof ElasticQuery) {
+      document.cookie = `crafterSite=${ config.site }`;
```
Comment (rart) [inline packages/search/src/search-service.ts]: Check if we should add path or other cookie directives

Comment (jvega190) [reply packages/search/src/search-service.ts]: actually we don't need this cookie anymore, proceeding to remove the line

**Pull request merged**